fee_allocator/helpers.py

```diff
@@ -321,22 +287,22 @@ def fetch_token_price_balgql_timerange(
         headers={"chainId": CHAIN_TO_CHAIN_ID_MAP[chain]},
     )
     client = Client(transport=transport, fetch_schema_from_transport=True)
-    query = gql(BAL_GQL_QUERY.format(token_addr=token_addr.lower()))
+    query = gql(
+        BAL_GQL_QUERY.format(
+            token_addr=token_addr.lower(), upper_chain_name=chain.upper()
+        )
+    )
     result = client.execute(query)
+    prices = result["tokenGetHistoricalPrices"][0]["prices"]
+    # Filter results so they are in between start_date and end_date timestamps
     # Sort result by timestamp desc
-    result["tokenGetPriceChartData"].sort(key=lambda x: x["timestamp"], reverse=True)
+    time_sorted_prices = sorted(prices, key=lambda x: int(x["timestamp"]), reverse=True)
     # Filter results so they are in between start_date and end_date timestamps
     result_slice = [
         item
-        for item in result["tokenGetPriceChartData"]
-        if end_date_ts >= item["timestamp"] >= start_date_ts
+        for item in time_sorted_prices
+        if end_date_ts >= int(item["timestamp"]) >= start_date_ts
     ]
-    if token_addr.lower() == "0x6dbf2155b0636cb3fd5359fccefb8a2c02b6cb51":
```

Comment:
Note I found these here.  Should check them in test run results and think about what this means.  As I recall they were hardcoded in because the API wasn't returning prices and it was crashing the job many moons ago right as shak was going.